### ISSUE TITLE
WiserAssertions.withContentContains fails with nested multipart parts

### ISSUE THREAD

**Issue by jonjo-manywho** [kemitix/wiser-assertions#1]
Hi,

I've noticed that when a message contains nested multipart parts, the `getMimeMessageBody` method fails to convert the message to a `String` correctly, and the assertion ends up looking at a value of `javax.mail.internet.MimeMultipart@2eada095`.

The plain message content and how it's generated is as follows:

```java
// I'm using org.codemonkey.simplejavamail:simple-java-mail:2.5 for this bit

final Email email = new Email();
email.addRecipient("Jonjo McKay", "jonjo.mckay@manywho.com", Message.RecipientType.TO);
email.setFromAddress("ManyWho", "no-reply@manywho.com");
email.setSubject("New activity");
email.setText("Hi Jonjo McKay,\n" +
        "\n" +
        "A new message was just posted in a stream you follow on ManyWho. The message was:\n" +
        "\n" +
        "Lance Drake Mandrell: \"This is a test message\"\n" +
        "\n" +
        "Join the flow at https://flow.manywho.com to read the stream and reply.\n" +
        "\n" +
        "ManyWho Email Bot");

mailer.sendMail(email);
```

```
Received: from trevor (localhost.localdomain [127.0.0.1])
        by trevor
        with SMTP (SubEthaSMTP 3.1.7) id IJTVO8YF
        for jonjo.mckay@manywho.com;
        Mon, 25 Jan 2016 11:15:36 +0000 (GMT)
Date: Mon, 25 Jan 2016 11:15:36 +0000 (GMT)
From: ManyWho <no-reply@manywho.com>
To: Jonjo McKay <jonjo.mckay@manywho.com>
Message-ID: <62259848.3.1453720536622@trevor>
Subject: New activity
MIME-Version: 1.0
Content-Type: multipart/mixed; 
	boundary="----=_Part_0_836429443.1453720536568"

------=_Part_0_836429443.1453720536568
Content-Type: multipart/related; 
	boundary="----=_Part_1_543941625.1453720536595"

------=_Part_1_543941625.1453720536595
Content-Type: multipart/alternative; 
	boundary="----=_Part_2_417869968.1453720536595"

------=_Part_2_417869968.1453720536595
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 7bit

Hi Jonjo McKay,

A new message was just posted in a stream you follow on ManyWho. The message was:

Lance Drake Mandrell: "This is a test message"

Join the flow at https://flow.manywho.com to read the stream and reply.

ManyWho Email Bot
------=_Part_2_417869968.1453720536595--

------=_Part_1_543941625.1453720536595--

------=_Part_0_836429443.1453720536568--
```

**Comment by kemitix** [comment]
I've created a draft test to try and catch this. I'm not familiar with `simple-java-mail` though; I'm getting an error:

```
org.codemonkey.simplejavamail.MailException: Generic error: Invalid protocol: null
	at net.kemitix.wiser.assertions.Issue1Test.shouldParseNestedMultiPartEmails(Issue1Test.java:32)
Caused by: javax.mail.NoSuchProviderException: Invalid protocol: null
	at net.kemitix.wiser.assertions.Issue1Test.shouldParseNestedMultiPartEmails(Issue1Test.java:32)
```

Could you have a look at the branch `issue-1-nested-multipart-mails` and send a pull request to complete the failing test? (Or even a fix if you feel inclined)

**Comment by jonjo-manywho** [comment]
Sure, I'll fork and send a pull request!

**Comment by kemitix** [comment]
Thanks for the PR. Fixed in 0.3.0 which is on it's way to the public repos now.

**Comment by jonjo-manywho** [comment]
Awesome! :+1:
